src/rdkafka.c

```diff
@@ -2661,6 +2666,8 @@ rd_kafka_t *rd_kafka_new(rd_kafka_type_t type,
          * that belong to rk_conf and thus needs to be cleaned up.
          * Legacy APIs, sigh.. */
         if (app_conf) {
+                if (group_remote_assignor_override)
+                        rd_free(group_remote_assignor_override);
```

Comment:
This can be outside of `if (app_conf)` condition as we are already doing a NULL check here.

Reply:
if not `app_conf `the property if freed in `rd_kafka_destroy_internal `so it would be a double free, here it's setting memory to zero later
`memset(&rk->rk_conf, 0, sizeof(rk->rk_conf));`
so they're not freed by `rd_kafka_destroy_internal ` and the properties that belonged to `app_conf `are still valid pointers that are freed by the app.